File: coding-test/프로그래머스/level1/로또의최고순위와최저순위/로또의최고순위와최저순위.py
def solution(lottos, win_nums):
    
    rank = [6,6,5,4,3,2,1]
    count = 0
    zero_count = 0
    
    for num in lottos:
        if num == 0:
            zero_count += 1
            continue
            
        for win_num in win_nums:
            if num == win_num:
                count += 1
    
    high_score = rank[zero_count + count]
    low_score = rank[count]
    
    return [high_score, low_score]

File: coding-test/프로그래머스/level1/로또의최고순위와최저순위/로또의최고순위와최저순위.js
function solution(lottos, winNums) {
  let count = 0;
  let zero = 0;

  lottos.forEach((num) => {
    if (winNums.includes(num)) count++;
    if (num === 0) zero++;
  });

  let high = 7 - (count + zero);
  let low = 7 - count;

  if (zero === 6) low--;
  if (count === 0 && zero === 0) {
    high--;
    low--;
  }

  return [high, low];
}
